### PR TITLE
chore: bazel debian artifacts are published

### LINKED ISSUE #14121: feat: dev label on AGW package
### Environment

- Version: Next after 1.8
- Affected Component: Access Gateway

#### Description

When a deployer has built their own magma package for installation in the AGW, they have no way to tell whether the built package is for dev or production. As a result, they may accidentally deploy a debug build on a production server, creating a security risk. A deployer should be able to check the build type.

This came up in the context of [Document secure deployment for "Build and install a magma package in AGW"](https://github.com/magma/security/issues/135), part of the epic [Documentation: write guide to secure deployment](https://github.com/magma/security/issues/104). This issue is a blocker for both of those.

#### Additional context

In the documentation [Build and install a magma package in AGW](https://docs.magmacore.org/docs/lte/build_install_magma_pkg_in_agw) (source: [build_install_magma_pkg_in_agw.md](https://github.com/magma/magma/blob/master/docs/readmes/lte/build_install_magma_pkg_in_agw.md)), 

- Step 3 is "Build and create deb package", with instructions on how to create a dev build or production build
- Step 9 is “verify the installed version”. But the command given does not show whether the build is for development or production

How can a user verify that the build installed was built for release and not dev?

#### Description

Add the build type to the description in the package, so that: 

```
$ apt show magma | grep Description
Description: Magma Access Gateway
-> Description: Magma Access Gateway - dev build
```

#### Alternative approaches

release vs dev only affects the cmake build-type, i.e., it's only relevant for the c/c++ services. I am no c/c++ expert, but iiuc: at the end this affects the build flags used. 

There are (default) flag collections in variables suffixed by _RELWITHDEBINFO (fab release) and _DEBUG (fab dev).

Note that we change the default flag collections differently in our different c/c++ services.

We do not record our gcc flags in the binary, i.e., I think there is no easy way to check this (e.g. `readelf -p .GCC.command.line /usr/local/bin/connectiond` is empty).

There might be some indicators ...
- the size of the artifacts as we use different optimization options
- the sanitizer features (dev is using address and release is using leak) -> `nm -an /usr/local/bin/connectiond | grep asan vs nm -an /usr/local/bin/connectiond | grep lsan`

#### Desired timelines

We should aim to ship this in the next release after 1.8.

### LINKED ISSUE #14299: The magma and sctpd bazel deb artifacts are pushed to the artifactory
* how to handle package name? if we have make and bazel created packages?
* ideas:
  * use 1.9.0 for bazel and keep make at 1.8.0 <--team decision for now
    * integ test make deb would still use 1.8.0 in CI, but 1.9 (bazel) in manual runs
  * use different package name, e.g., magmab, sctpdb
    * needs to be switched later
  * disable make debian creation
    * integ test make deb would also use bazel package

#### requirements
* do this in the bazel.yml workflow - only on master merges
* artifacts are build with production config
* file name is changed to same pattern as with make `~ magma_1.9.0_<timestamp>_<short_commit_hash>.deb`
* version is set to same pattern as in make `1.9.0_<timestamp>_<short_commit_hash>`
* in general: change version to 1.9.0
* if possible, already do the double push to both artifactories - if not, create follow-up to do this when the new artifactory is ready

#### create follow-ups
* use this in bazel magma deb workflow (so docker setup can be removed from workflow)
  * this is: bazel.yml package build triggers the workflow on master 
* (already exists) switch make magma deb in ci dashboard to bazel magma deb
* remove make magma deb workflow

### PR DESCRIPTION
**Pull request by nstng**
## Summary

This change will allow publishing magma and sctpd debian packages created with bazel to the magmacore artifactory. For this the following was implemented (corresponds to the commits):
1. Create a bazel target that builds the debian packages and afterwards changes the meta info of the artifacts (version, description, post-install script) and renames the files. This target is for manual execution - CI and local builds. Note that the version for these artifacts is set to 1.9.0 - the make artifacts still use 1.8.0 (see also https://magmacore.slack.com/archives/C0301NRRL7K/p1667298024205399).
2. +5. In the bazel.yml workflow add a step where the packages are published to the artifactory (only for merges on master). This is mostly reused from ~~build_all.yml~~ build_magma_dep.yml.
3. Make sure that a locally created magma_deb VM (or executing the [lte make debian integ tests](https://github.com/magma/magma/actions/workflows/lte-integ-test-magma-deb.yml) on a fork still use the latest magm artifact created with make.
4. Adapt documentation for manually building magma/sctpd debian artifacts.

Will also close #14121.

## Test Plan

build locally based on the documentation.

Run in CI (only uploads on master). Here the output for a modified run where the upload step was enabled but only produced logging (https://github.com/magma/magma/actions/runs/3371605060/jobs/5594061360):
```
Pushing package magma-sctpd_1.9.0-1667325390-acda59fe_amd64.deb to JFROG artifiactory: https://artifactory.magmacore.org/artifactory/debian-test/pool
Pushing package magma_1.9.0-1667325390-acda59fe_amd64.deb to JFROG artifiactory: https://artifactory.magmacore.org/artifactory/debian-test/pool

```

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/release/update_deb_meta_info.sh.template [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+###############################################################################
+# FUNCTION DECLARATIONS
+###############################################################################
+
+update_version() {
+    local INPUT_FILE_NAME=$1
+
+    rm -rf "${WORK_FOLDER}"
+    mkdir "${WORK_FOLDER}"
```
Comment (Neudrino) [inline lte/gateway/release/update_deb_meta_info.sh.template]: Should this have a `-p` option, in case a subfolder is given?

File: lte/gateway/release/update_deb_meta_info.sh.template [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    cd "${WORK_FOLDER}"
+
+    echo "Updating meta data in ${INPUT_FILE_NAME} ..."
```
Comment (Neudrino) [inline lte/gateway/release/update_deb_meta_info.sh.template:28]: The echo commands below always start with 2 white-space in the beginning. However, here they are missing. Is this intended?

File: lte/gateway/release/update_deb_meta_info.sh.template [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+VERSION=REPLACE_ME # set by genrule
+MAGMA_INPUT_FILE_NAME=REPLACE_ME # set by genrule
+SCTPD_INPUT_FILE_NAME=REPLACE_ME # set by genrule
+DEV_BUILD=true # can be overridden by genrule
+
+WORK_FOLDER=/tmp/repackage_workfolder
+RESULT_FOLDER=/tmp/packages
+
+BUILD_PATH="${MAGMA_ROOT}/bazel-bin/lte/gateway/release"
+REPO="${MAGMA_ROOT}"
```
Comment (Neudrino) [inline lte/gateway/release/update_deb_meta_info.sh.template]: Why is there an extra variable necessary here? Could we not just use `${MAGMA_ROOT}` instead?

File: lte/gateway/release/update_deb_meta_info.sh.template [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+git config --global --add safe.directory "${MAGMA_ROOT}"
+
+HASH=$(git -C "${REPO}" rev-parse HEAD)
+COMMIT_COUNT=$(git -C "${REPO}" rev-list --count HEAD)
+HASH_SHORT=${HASH::8}
+TIMESTAMP=$(date +%s)
+VERSION_SUFFIX="${TIMESTAMP}-${HASH_SHORT}"
+
+rm -rf /tmp/packages
+mkdir /tmp/packages
```
Comment (Neudrino) [inline lte/gateway/release/update_deb_meta_info.sh.template]: - The `work_folder` is created inside the function. This one outside. This seems like an inconsistency. Is there a reason for this difference?
- `mkdir -p` might be more secure, even though, it might not be necessary on most unix systems in this specific case.

File: .github/workflows/bazel.yml [hunk 5]
```diff
@@ -291,9 +291,37 @@ jobs:
...
               --profile=Bazel_build_package_profile
+            mkdir packages
+            mv /tmp/packages/*.deb packages
+      - name: Publish magma and sctpd
+        if: github.event_name == 'push' && github.repository_owner == 'magma'
+        env:
+          ARTIFACTORY_URL: "https://artifactory.magmacore.org/artifactory/debian-test/pool"
+          DEBIAN_META_INFO: ";deb.distribution=focal-ci;deb.component=main;deb.architecture=amd64"
+        run: |
+          cd packages
```
Comment (Neudrino) [inline .github/workflows/bazel.yml]: Using [`working-directory:`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun) instead, might be cleaner.

File: .github/workflows/bazel.yml [hunk 6]
```diff
@@ -291,9 +291,37 @@ jobs:
...
+          for package in `ls -a1 *.deb`
+          do
+            echo "Pushing package ${package} to JFROG artifiactory: ${ARTIFACTORY_URL}"
+            HTTP_RESPONSE=$(curl --silent --write-out "HTTPSTATUS:%{http_code}" -uci-bot:${{ secrets.JFROG_CIBOT_APIKEYS }} -XPUT "${ARTIFACTORY_URL}/focal-ci/${package}${DEBIAN_META_INFO}" -T $package)
```
Comment (Neudrino) [inline .github/workflows/bazel.yml]: - It might be easier to use the jfrog cli. We already do so: https://github.com/magma/magma/blob/5a72ab813e340ccd43ca6766362d96fb229d3b90/.github/workflows/build_magma_dep.yml#L80-L88
- jfrog cli or not, it would be nice, if we could mulit-line those many command line arguments for better readability.

File: .github/workflows/bazel.yml [hunk 7]
```diff
@@ -291,9 +291,37 @@ jobs:
...
+            URI=$(echo $HTTP_BODY | jq -r '.uri')
+            if [[ "$URI" != "null" ]]; then
+              ARTIFACTS=$(echo $ARTIFACTS | jq --arg uri $URI '.packages += [$uri]')
+            else
+              PUBLISH_ERROR="true"
+            fi
+            # extract and check status
+            HTTP_STATUS=$(echo $HTTP_RESPONSE | tr -d '\n' | sed -e 's/.*HTTPSTATUS://')
+            if [[ "$HTTP_STATUS" != "2"* ]]; then
+              PUBLISH_ERROR="true"
```
Comment (Neudrino) [inline .github/workflows/bazel.yml]: Is this `PUBLISH_ERROR` checked somewhere?

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md [hunk 8]
```diff
@@ -5,11 +5,15 @@ hide_title: true
 ---
 # Build and install a magma package in AGW
 
-**Description:** Purpose of this document is describe how to build and install a magma package in AGW.
+## Description
 
-**Environment:** AGW
+Purpose of this document is describe how to build and install a magma package in AGW.
```
Comment (Neudrino) [inline docs/readmes/lte/build_install_magma_pkg_in_agw.md]: `Purpose of this document is _to_ describe...`

File: .github/workflows/bazel.yml [hunk 9]
```diff
@@ -291,9 +291,37 @@ jobs:
           run: |
             cd /workspaces/magma
             bazel/scripts/remote_cache_bazelrc_setup.sh "${{ env.CACHE_KEY }}" "${{ env.REMOTE_DOWNLOAD_OPTIMIZATION }}" "${{ secrets.BAZEL_REMOTE_PASSWORD }}"
-            bazel build lte/gateway/release:sctpd_deb_pkg lte/gateway/release:magma_deb_pkg \
+            bazel run //lte/gateway/release:release_build \
```
Comment (jheidbrink) [inline .github/workflows/bazel.yml:294]: We might let `release_build` take the output path as parameter, so we don't have to know that it's `/tmp/packages`

File: .github/workflows/bazel.yml [hunk 10]
```diff
@@ -291,9 +291,37 @@ jobs:
...
+      - name: Publish magma and sctpd
+        if: github.event_name == 'push' && github.repository_owner == 'magma'
+        env:
+          ARTIFACTORY_URL: "https://artifactory.magmacore.org/artifactory/debian-test/pool"
+          DEBIAN_META_INFO: ";deb.distribution=focal-ci;deb.component=main;deb.architecture=amd64"
+        run: |
+          cd packages
+          for package in `ls -a1 *.deb`
+          do
+            echo "Pushing package ${package} to JFROG artifiactory: ${ARTIFACTORY_URL}"
```
Comment (sebathomas) [inline .github/workflows/bazel.yml]:
```suggestion
            echo "Pushing package ${package} to JFROG artifactory: ${ARTIFACTORY_URL}"
```

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md [hunk 11]
```diff
@@ -74,3 +78,80 @@ hide_title: true
...
+
+1. **Clone or Update  your local repository**.
+    In any machine (i.e. your computer) clone the magma repo.
+
+    ```bash
+    git clone https://github.com/magma/magma.git
+    ```
+
+2. **Install prerequisites**.
+    Make sure you have installed all the tools specified in the prerequisites <https://magma.github.io/magma/docs/basics/prerequisites#prerequisites>
```
Comment (Neudrino) [inline docs/readmes/lte/build_install_magma_pkg_in_agw.md:28]: For now, the official documentation is still at https://docs.magmacore.org/, even though the deployment to that documentation is broken atm.
It is true, that the Github pages version is up to date and we should discuss if we want to switch to it as main documentation reference.
However, until a decision is reached regarding the deployment infra of the docs, I would keep it consistent and use the above mentioned reference.

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md [hunk 12]
```diff
@@ -74,3 +78,80 @@ hide_title: true
     ```bash
     apt show magma
     ```
+
+## Steps with Bazel
```
Comment (jheidbrink) [inline docs/readmes/lte/build_install_magma_pkg_in_agw.md]: Can we already replace "steps" with the new "steps with Bazel" section?

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md [hunk 13]
```diff
@@ -74,3 +78,80 @@ hide_title: true
...
+
+    There should be two packages named `magma_1.9.XXX` and `magma-sctpd_1.9.XXX` (for v1.9 versions).
+
+5. **Download the package**.
+    You can download the files to your computer from the vagrant machine. To do so, you can install a vagrant plugin in your computer and then download the package from the VM to your computer with the following commands:
+
+    ```bash
+    vagrant plugin install vagrant-scp
+    vagrant scp magma: /tmp/packages/<deb_package>
+    ```
```
Comment (Neudrino) [inline docs/readmes/lte/build_install_magma_pkg_in_agw.md]: Would it not be easier to just move them to a folder, which is mounted from the host? E.g. the repo?

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md (hunk 12 above)
Comment (nstng) [reply docs/readmes/lte/build_install_magma_pkg_in_agw.md]: The "steps" section is for 1.8 where this change is not supported.

File: .github/workflows/bazel.yml [hunk 14]
```diff
@@ -291,9 +291,37 @@ jobs:
...
+            mkdir packages
+            mv /tmp/packages/*.deb packages
+      - name: Publish magma and sctpd
+        if: github.event_name == 'push' && github.repository_owner == 'magma'
+        env:
+          ARTIFACTORY_URL: "https://artifactory.magmacore.org/artifactory/debian-test/pool"
+          DEBIAN_META_INFO: ";deb.distribution=focal-ci;deb.component=main;deb.architecture=amd64"
+        run: |
+          cd packages
+          for package in `ls -a1 *.deb`
```
Comment (jheidbrink) [inline .github/workflows/bazel.yml]: nitpick: `for package in *.deb` should also work

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md (hunk 12 above)
Comment (jheidbrink) [reply docs/readmes/lte/build_install_magma_pkg_in_agw.md]: I know, I was suggesting to not keep that around anymore. So someone who looks up how to build a deb package today will find the Bazel version.

File: .github/workflows/bazel.yml (hunk 9 above)
Comment (jheidbrink) [reply .github/workflows/bazel.yml:294]: or just put them inside the working directory

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md [hunk 15]
```diff
@@ -74,3 +78,80 @@ hide_title: true
...
+4. **Locate the packages**.
+    Once the above command finished you can find the packages inside the VM:
+
+    ```bash
+    cd /tmp/packages
+    ```
+
+    There should be two packages named `magma_1.9.XXX` and `magma-sctpd_1.9.XXX` (for v1.9 versions).
+
+5. **Download the package**.
```
Comment (jheidbrink) [inline docs/readmes/lte/build_install_magma_pkg_in_agw.md]: I would probably not include this in the documentation. It's not very specific to the build process. Whoever follows this will need a basic understanding of how to copy files between machines anyway (for example in the next section, "upload to agw"). And this is somewhat opinionated, some people would just use `scp` without Vagrant plugin, some would rather copy it to $MAGMA_ROOT etc.

File: .github/workflows/bazel.yml (hunk 9 above)
Comment (nstng) [reply .github/workflows/bazel.yml:294]: discussed offline - will stay as is for now

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md (hunk 12 above)
Comment (nstng) [reply docs/readmes/lte/build_install_magma_pkg_in_agw.md]: done

File: lte/gateway/release/update_deb_meta_info.sh.template (hunk 2 above)
Comment (nstng) [reply lte/gateway/release/update_deb_meta_info.sh.template:28]: Yes, it is - so that follow up output is associated to this line. Sample output:
```
Updating meta data in magma_1.9.0-VERSION-SUFFIX_amd64.deb ...
  Updating version "1.9.0-VERSION-SUFFIX" to "1.9.0-1667398648-cd2b07e9"
  Updating description to development build "false"
  Updating "COMMIT_HASH" in magma to "magma@1.9.0.11665-cd2b07e9"
  Renaming "magma_1.9.0-VERSION-SUFFIX_amd64.deb" to "magma_1.9.0-1667398648-cd2b07e9_amd64.deb"
  File written to /home/vscode/magma/magma_1.9.0-1667398648-cd2b07e9_amd64.deb
Updating meta data in magma-sctpd_1.9.0-VERSION-SUFFIX_amd64.deb ...
  Updating version "1.9.0-VERSION-SUFFIX" to "1.9.0-1667398648-cd2b07e9"
  Updating description to development build "false"
  Renaming "magma-sctpd_1.9.0-VERSION-SUFFIX_amd64.deb" to "magma-sctpd_1.9.0-1667398648-cd2b07e9_amd64.deb"
```

File: lte/gateway/release/update_deb_meta_info.sh.template (hunk 4 above)
Comment (nstng) [reply lte/gateway/release/update_deb_meta_info.sh.template]: The result folder will get the created debian files from the function - the function is called two times. For each run the workfolder needs to be cleaned up.
-p: :+1:

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md (hunk 8 above)
Comment (voisey) [reply docs/readmes/lte/build_install_magma_pkg_in_agw.md]: I would prefer "The purpose is..." too

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md [hunk 16]
```diff
@@ -74,3 +78,80 @@ hide_title: true
     ```bash
     apt show magma
     ```
+
+## Steps with Bazel
+
+This is not supported for 1.8, but is possible on the development branch and will replace the documentation above in future versions.
+
+1. **Clone or Update  your local repository**.
```
Comment (voisey) [inline docs/readmes/lte/build_install_magma_pkg_in_agw.md]:
```suggestion
1. **Clone or Update your local repository**.
```

File: .github/workflows/bazel.yml (hunk 7 above)
Comment (nstng) [reply .github/workflows/bazel.yml]: :+1: will be re-worked

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md (hunk 13 above)
Comment (nstng) [reply docs/readmes/lte/build_install_magma_pkg_in_agw.md]: will be removed. Should be expected basic understanding.

File: lte/gateway/release/update_deb_meta_info.sh.template (hunk 1 above)
Comment (nstng) [reply lte/gateway/release/update_deb_meta_info.sh.template]: changed to `mktemp`

File: lte/gateway/release/update_deb_meta_info.sh.template (hunk 3 above)
Comment (nstng) [reply lte/gateway/release/update_deb_meta_info.sh.template]: removed

File: .github/workflows/bazel.yml (hunk 5 above)
Comment (nstng) [reply .github/workflows/bazel.yml]: added

File: .github/workflows/bazel.yml (hunk 6 above)
Comment (nstng) [reply .github/workflows/bazel.yml]: will have a look

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md (hunk 8 above)
Comment (nstng) [reply docs/readmes/lte/build_install_magma_pkg_in_agw.md]: changed

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md (hunk 16 above)
Comment (nstng) [reply docs/readmes/lte/build_install_magma_pkg_in_agw.md]: changed

**Comment by nstng** [comment]
Set to draft for jfrog cli experiments

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md [hunk 17]
```diff
@@ -5,71 +5,79 @@ hide_title: true
...
+
+Note that the following is supported starting with magma v1.9. For older versions see respective documentation.
+
+1. **Clone or Update your local repository**.
+    In any machine (i.e. your computer) clone the magma repo (and switch to the branch you want to build).
 
     ```bash
     git clone https://github.com/magma/magma.git
-    git checkout v1.8
+    # git switch <branch>
```
Comment (Neudrino) [inline docs/readmes/lte/build_install_magma_pkg_in_agw.md]: Could we also completely delete this?
I think commented out, it is not of much help. In the line if commands a `cd` into the folder would be missing anyway.
Imho, we should be able to expect the knowledge of git usage.

Comment (nstng) [reply docs/readmes/lte/build_install_magma_pkg_in_agw.md]: removed

**Comment by nstng** [comment]
ready for review again - jfrog cli output: https://github.com/magma/magma/actions/runs/3379084377/jobs/5610181245
```
16:00:57 [Info] [Thread 0] [Dry run] Uploading artifact: packages/magma_1.9.0-1667404851-d7ad6c30_amd64.deb
16:00:57 [Info] [Thread 1] [Dry run] Uploading artifact: packages/magma-sctpd_1.9.0-1667404851-d7ad6c30_amd64.deb
{
  "status": "success",
  "totals": {
    "success": 2,
    "failure": 0
  },
  "files": [
    {
      "source": "packages/magma-sctpd_1.9.0-1667404851-d7ad6c30_amd64.deb",
      "target": "[https://artifactory.magmacore.org:443/artifactory/debian-test/pool/focal-ci/magma-sctpd_1.9.0-1667404851-d7ad6c30_amd64.deb](https://artifactory.magmacore.org/artifactory/debian-test/pool/focal-ci/magma-sctpd_1.9.0-1667404851-d7ad6c30_amd64.deb)",
      "sha256": ""
    },
    {
      "source": "packages/magma_1.9.0-1667404851-d7ad6c30_amd64.deb",
      "target": "[https://artifactory.magmacore.org:443/artifactory/debian-test/pool/focal-ci/magma_1.9.0-1667404851-d7ad6c30_amd64.deb](https://artifactory.magmacore.org/artifactory/debian-test/pool/focal-ci/magma_1.9.0-1667404851-d7ad6c30_amd64.deb)",
      "sha256": ""
    }
  ]
}
```

File: .github/workflows/bazel.yml (hunk 6 above)
Comment (nstng) [reply .github/workflows/bazel.yml]: seems to work nice - thank you for the suggestion

File: .github/workflows/bazel.yml [hunk 18]
```diff
@@ -291,9 +291,31 @@ jobs:
...
+      - name: Set dry run environment variable
+        if: ${{ ! ( github.event_name == 'push' && github.repository_owner == 'magma' ) }}
+        run: |
+          echo "IS_DRY=--dry-run" >> $GITHUB_ENV
+      - name: Publish debian packages
+        env:
+          ARTIFACTORY_URL: https://artifactory.magmacore.org:443/artifactory/
+          DEBIAN_META_INFO: deb.distribution=focal-ci;deb.component=main;deb.architecture=amd64
+        run: |
+          jf rt u \
```
Comment (jheidbrink) [inline .github/workflows/bazel.yml]: I would use the long command "upload" instead of "u" here

File: docs/readmes/lte/build_install_magma_pkg_in_agw.md [hunk 19]
```diff
@@ -5,71 +5,78 @@ hide_title: true
...
-8. **Restart the magma services**
+7. **Restart the magma services**
 
     ```bash
     sudo service magma@* stop
     sudo service magma@magmad restart
     ```
 
-9. You can **verify the installed version** with
+8. You can **verify the installed version** with
```
Comment (Neudrino) [inline docs/readmes/lte/build_install_magma_pkg_in_agw.md:79]: _Only FYI - you do not need to change back or do yet another commit!:_
Changing this is usually not necessary when markdown is correctly translated. In fact, Markdown numbered lists should usually only be numbered with `1.` for all items to avoid those changes and confusion.
E.g. a "randomly" numbered list and its rendering:
```markdown
1. First
1. Second
2. Third
43. Fourth
1. Fiveth
```
Results in:
1. First
1. Second
2. Third
43. Fourth
1. Fiveth

File: lte/gateway/release/BUILD.bazel [hunk 20]
```diff
@@ -10,9 +10,8 @@
 # limitations under the License.
 
 """
-Artifacts in this package should be build with the production
-configuration:
-bazel build lte/gateway/release:sctpd_deb_pkg --config=production
+Definitions of the magma and sctpd debian artifacts.
+See "//lte/gateway/release:release_build" on how artifacts can be created.
```
Comment (jheidbrink) [inline lte/gateway/release/BUILD.bazel]: nitpick:  Given that that target lives in this package, we could just reference the package-local name ("look at the the comment in the "release_build" target for information on how to build the Magma deb package")

**Review by Neudrino — APPROVED** [review]

**Review by jheidbrink — APPROVED** [review]

File: lte/gateway/release/BUILD.bazel (hunk 20 above)
Comment (nstng) [reply lte/gateway/release/BUILD.bazel]: :+1: changed

File: .github/workflows/bazel.yml (hunk 18 above)
Comment (nstng) [reply .github/workflows/bazel.yml]: :+1: changed

**Review by sebathomas — APPROVED** [review]
